.github/workflows/_test_slurm_pyxis.yaml

```diff
@@ -0,0 +1,248 @@
+name: ~test multi-node jobs via SLURM+Pyxis
+
+on:
+  workflow_call:
+    secrets:
+      SSH_PRIVATE_KEY:
+        required: true
+        description: SSH private key for accessing the SLURM login node
+      SLURM_LOGIN_USER:
+        required: true
+        description: Username for the SLURM login node
+      CONTAINER_REGISTRY_TOKEN:
+        required: true
+        description: Token for accessing the container registry
+    inputs:
+      NAME:
+        type: string
+        description: Name of the test case
+        required: true
+      ARTIFACT_NAME:
+        type: string
+        description: Name of the artifact zip file
+        required: false
+        default: ''
+      SLURM_LOGIN_HOSTNAME:
+        type: string
+        description: Hostname of the SLURM login node
+        required: true
+      SLURM_SCRATCH_PATH:
+        type: string
+        description: Path to the scratch directory on the SLURM cluster
+        required: true
+      NODES:
+        type: number
+        description: Number of nodes to request
+        required: true
+      GPUS_PER_NODE:
+        type: number
+        description: Number of GPUs per node to request
+        required: true
+      NTASKS:
+        type: number
+        description: Number of tasks to run
+        required: true
+      NTASKS_PER_NODE:
```

Comment:
Can all these parameters be a part of SLURM_SCRIPT?

Reply:
Probably not, because the workflow will insert preambles and other stuff before SLURM_SCRIPT, while #!SBATCH directives must appear at top of the script. The two conditions cannot be simultaneously satisfied here.